### ISSUE TITLE
Multihand tracking using GPU in Ubuntu terminates abruptly with signal SIGSEGV (Address boundary error)

### ISSUE THREAD

**Issue by prantoran** [google/mediapipe#833]
Hello ^_^

I am working on restructuring the example code with separate Bazel build script and header files to contain application-specific logic. I am sending the OpenCV matrix as an input just as in the GPU example. But instead of the matrix derived from GPU, I am sending the original OpenCV matrix augmented with additional drawings. I am using an integrated GPU, not a dedicated one, but the fps is pretty good.

The problem is that for the GPU example, the program sometimes crashes with SIGSEGV signal. And the frequency of SIGSEGV increases with each additional increase in code complexity. I was wondering whether it is a recurring problem and if so then how to fix the crashes.

**Comment by prantoran** [comment]
I also get this error message:

`
*printout of graph config*
I20200623 14:43:38.217983 14650 multihand_gpu.cc:118] Initialize the calculator graph.
I20200623 14:43:38.223321 14650 multihand_gpu.cc:122] Initialize the GPU.
I20200623 14:43:38.239187 14650 gl_context_egl.cc:158] Successfully initialized EGL. Major : 1 Minor: 4
I20200623 14:43:38.243221 14663 gl_context.cc:324] GL version: 3.1 (OpenGL ES 3.1 Mesa 20.0.7)
I20200623 14:43:38.243337 14650 multihand_gpu.cc:128] Initialize the camera or load the video.
I20200623 14:43:38.525593 14650 multihand_gpu.cc:152] Start running the calculator graph.
I20200623 14:43:38.526007 14650 multihand_gpu.cc:159] Start grabbing and processing frames.
INFO: Created TensorFlow Lite delegate for GPU.
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
fish: “GLOG_logtostderr=1 bazel-bin/Pa…” terminated by signal SIGABRT (Abort)

`

**Comment by mcclanahoochie** [comment]
which gpu do you have? how much ram does your gpu have? 
the hand tracking examples use quite a bit of gpu memory (something that is being worked on resolving).

does the unmodified example show this issue?

do the other gpu examples show the same issue? (face detection, single hand tracking, face mesh)?

>  increases with each additional increase in code complexity
where is this new code?... in the demo runner, or are you adding/modifying calculators?
how is new cpu/gpu memory being managed?

**Comment by prantoran** [comment]
My GPU is a pretty low-config integrated GPU with 2GB ram. The exact model is `Mars [Radeon HD 8670A/8670M/8750M]`. The bare hand-tracking example with GPU works smoothly without any modification. I did not try other examples yet.

My new code is separate from the MediaPipe framework. I did not use any calculators. Sequentially without any multithreading, I fed the OpenCV input image matrix to the MediaPipe graph, extracted the landmarks, and modified the original OpenCV input based on the landmarks and fed back to the output. I included MediaPipe inside a class in a separate header file.

One observation is that when I had debugged messages to console for each iteration, there have not been any crashes. Maybe, the crashes are due to mismatch in concurrent processes or pile up of objects in memory.
